### DISCUSSION TITLE
How can I use katex from my md file?

### DISCUSSION THREAD

**Discussion by manuelgomezrico** [KaTeX/KaTeX#2781] [Q&A]
Hi
I have writed the following formula in my md file from visual studio code:

$$RRMSE= \sqrt{\dfrac{1}{n} * \displaystyle\sum_{i=1}^n\dfrac{(y_i-\tilde{\bar{y_i})^2}}{y_i}}$$

And I see it in visual code very well
![138468F4-59C2-487D-A260-915A3404B005_4_5005_c](https://user-images.githubusercontent.com/6414377/107251566-c4f27600-6a34-11eb-8b4b-0f1e3da7888e.jpeg)

When I upload this rm file to project github, I see again:
$$RRMSE= \sqrt{\dfrac{1}{n} * \displaystyle\sum_{i=1}^n\dfrac{(y_i-\tilde{\bar{y_i})^2}}{y_i}}$$
I want to se the formula of image, could you help me,please?
![138468F4-59C2-487D-A260-915A3404B005_4_5005_c](https://user-images.githubusercontent.com/6414377/107251566-c4f27600-6a34-11eb-8b4b-0f1e3da7888e.jpeg)

Thanks

**Comment by edemaine** [comment]
Github's Markdown renderer does not support KaTeX.  I believe you can use Github Pages, which uses Jekyll, with suitable plugins, but there's no way to get READMEs etc. to directly render mathematics, I'm afraid.

**Reply by manuelgomezrico** [reply]
Ok, thanks for your answer 
So I'm sad because I did one formula so cool from visual studio code :(

**Reply by manuelgomezrico** [reply]
Hi again
I have the solution the problem
If you render the formula in img object, you can see the formula in radme.md without problem
Example:
<img src="https://render.githubusercontent.com/render/math?math=RRMSE=%20\sqrt{\dfrac{1}{n}%20*%20\displaystyle\sum_{i=1}^n\dfrac{(y_i-\tilde{\bar{y_i})^2}}{y_i}}">

The img object, in src attribute, you have to write the formula:
img src
"https://render.githubusercontent.com/render/math?math=RRMSE=%20\sqrt{\dfrac{1}{n}%20*%20\displaystyle\sum_{i=1}^n\dfrac{(y_i-\tilde{\bar{y_i})^2}}{y_i}}"
Then I can use this plugin in local (Visual Studio Code) and after I have to render it in img object (project Github)
Regards

**Reply by edemaine** [reply]
Thanks, I didn't know about https://gist.github.com/a-rodin/fef3f543412d6e1ec5b6cf55bf197d7b &mdash; quite interesting. Unfortunately, it doesn't work well in dark mode (I see black text on an almost-black background).

**Reply by manuelgomezrico** [reply]
if you use this attribute stype in object img like --> style="background: white;"
Is possible that you can have solution. What do you think about this?

**Comment by pdmosses** [comment]
I'm experimenting with using KaTeX in Markdown pages – see my [test site](https://pdmosses.github.io/just-the-docs-tests/docs/math/katex/). The KaTeX markup is rendered in the browser, so there is no need to upload any images; it works in dark mode too (try the Preview button on [this page](https://pdmosses.github.io/just-the-docs-tests/docs/math/katex/tests.html)).

The test site is built and hosted on Github Pages, and (as @edemaine mentioned) Github Pages uses Jekyll. By default, Jekyll uses [kramdown](https://kramdown.gettalong.org/) to convert (a variant of) Markdown to HTML, and you can use KaTeX markup in [kramdown's math blocks](https://kramdown.gettalong.org/syntax.html#math-blocks), without needing to escape special characters.

The only configuration required to enable use of KaTeX on the test site was to include the usual links and scripts in the `head` element of the relevant pages. For the Jekyll theme that I'm using ([Just the Docs](https://pmarsceill.github.io/just-the-docs/)) that involved [adding a single file](https://pdmosses.github.io/just-the-docs-tests/docs/math/katex/config.html), and adding `math: katex` in the front matter on pages with math blocks.

No extra plugins were required. You can also run Jekyll locally to build your site (independently of Github Pages) or use kramdown directly.
